README.md

```diff
@@ -1,17 +1,176 @@
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
 
 # 0 Abstract
 
-## Language
+[Interplanetary Linked Data (IPLD)](https://ipld.io/) is a consistent, highly general data model for content addressed linked data forming any DAG. This data model provides a convenient pivot between many serializations of the same information. Being a chained token model, UCANs can be very naturally expressed via IPLD. However, JWTs are not fully deterministic due to whitespace and key ordering. This specification defines an IPLD Schema for UCAN, and a canonical JWT serialization for compatibility with all other clients.
+
+# 1 Motivation
+
+The [core UCAN specification](https://github.com/ucan-wg/spec) is defined as a JWT to make it easily adoptable with existing tools, and as a common format for all implementations. There are many cases where different encodings are preferred for reasons such as compactness, machine-efficient formats, and so on. This specification outlines a format based on IPLD that can be deterministically encoded and decoded between many serialization formats, while still being able to encode as JWT for compatibility.
+
+# 2 IPLD Schema
+
+Unlike a JWT, the IPLD encoding of UCAN does not require separate header, claims, and signature fields.
+
+```ipldsch
+type UCAN struct {
+  v String
+
+  iss Principal
+  aud Principal
+  s Signature
+
+  att [Capability]
+  -- All proofs are links, however you could still inline proof
+  -- by using CID with identity hashing algorithm
+  prf [&UCAN]
+  exp Int
+
+  fct [Fact]
+  nnc optional String
+  nbf optional Int
+} representation map {
+  field fct default []
+  field prf default []
+}
+```
+
+## 2.1 Principal
+
+Principals MUST use [bytesprefix](https://ipld.io/docs/schemas/using/authoring-guide/#bytesprefix-unions-for-bytes) representation of [DID](https://www.w3.org/TR/did-core/)s. Primary [`did:key` method](https://w3c-ccg.github.io/did-method-key/) MUST use (more compact) key specific variant representation. Additional DID methods MAY use `DID` variant representation.
+
+``` ipldsch
+type Principal union {
+  -- Specification defines principals in terms of did:key's (that are multicodec
+  -- identifier for the public key type followed by the raw bytes of the key)
+  -- we represent those as raw public key bytes prefixed with public key multiformat
+  -- code.
+  | secp256k1  "0xe7"
+  | BLS12381G1 "0xea"
+  | BLS12381G2 "0xeb"
+  | Ed25519    "0xed"
+  | P256       "0x1200"
+  | P384       "0x1201"
+  | P512       "0x1202"
+  | RSA        "0x1205"
+  
+  -- To accomodate additional DID methods we represent those as UTF8 encoding
+  -- of the DID omitting `did:` prefix itself. E.g. `did:dns:ucan.xyz` can be
+  -- represented as [0xd1d, ...new TextEncoder().encode('dns:ucan.xyz')] bytes.
+  | DID "0x0d1d"
```

Comment:
@Gozala I changed this to break cleanly on bytes (even number of nibbles). It looks like that's the pattern in the multiformat table.